java/code/src/com/redhat/rhn/domain/channel/PublicChannelFamily.java

```diff
@@ -63,6 +83,7 @@ public boolean equals(final Object other) {
     @Override
     public int hashCode() {
         return new HashCodeBuilder()
+            .append(getId())
```

Comment:
Not sure if the "id" should be in the equals and hashcode method. I remember that this is a bad idea.

Reply:
In this specific use case it should not matter as the `id` is the `channel_family_id`. It's basically the only column of the table and it must already exists as it is a foreign key. But anyway you have a point, and there is strictly no reason to change the behaviour of the equals/hashcode in the first place. I will remove it.

Reply:
I removed the change to `equals()` and `hashcode()`.